lua/possession/commands.lua

```diff
@@ -49,13 +49,33 @@ local function get_session_names()
     if not cached_names then
         cached_names = {}
         for file, data in pairs(session.list()) do
-            cached_names[file] = data.name
+            cached_names[file] = { name = data.name, cwd = data.cwd }
         end
     end
     return cached_names
 end
 
-M.complete_session = complete_list(get_session_names)
+M.complete_session = complete_list(function()
+    return vim.tbl_map(function(s)
+        return s.name
+    end, get_session_names())
+end)
+
+M.cwd_complete_session = complete_list(function()
+    local cwds = vim.tbl_map(function(s)
+        return s.cwd
+    end, get_session_names())
+
+    local distinct_cwds = {}
+
+    for _, v in pairs(cwds) do
+        if not vim.tbl_contains(distinct_cwds, v) then
```

Comment:
If you know of a better or smarter way to get a distinct list of values in a table, let me know.

Your `utils.list_to_set` won't work here, this isn't a list.

Reply:
Interesting, I actually always thought `vim.tbl_map` only works on list-like tables, but I've checked the source code and it works on any table.

Anyway, your code is fine, but if you'd rather make use of `utils.list_to_set` then  you should be able to do:
```lua
local distinct_cwds = vim.tbl_keys(utils.list_to_set(vim.tbl_values(cwds)))
```